tree.go

```diff
@@ -462,6 +462,7 @@ walk: // Outer loop for walking the tree
 						for l := len(*skippedNodes); l > 0; {
```

Comment:
use `for l := len(*skippedNodes); l >0; l--)`?

maybe rename `l` to `length`, because `l` like `1`

Reply:
Much cleaner. Fixed :)